### PR TITLE
Update to jax 0.3.14

### PR DESCRIPTION
**Pull request by jkausrelay**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ channels:
   - openeye
 dependencies:
   - python=3.7
-  - jax=0.2.27
-  - jaxlib=0.1.75
```
Comment (proteneer) [inline environment.yml:8]: weird jaxlib got a huge bump it seems

Comment (jkausrelay) [reply environment.yml:8]: they decided to make them both use consistent version #s

Comment (proteneer) [reply environment.yml:8]: that just makes too much sense!

**Review by proteneer — APPROVED** [review]
lgtm

**Pull request merged**